tests/test_objective_funs.py

```diff
@@ -1284,31 +1284,38 @@ def test_derivative_modes():
     surf = FourierRZToroidalSurface()
     obj1 = ObjectiveFunction(
         [
-            PlasmaVesselDistance(eq, surf),
+            PlasmaVesselDistance(eq, surf, jac_chunk_size=1),
             MagneticWell(eq),
         ],
         deriv_mode="batched",
         use_jit=False,
     )
     obj2 = ObjectiveFunction(
         [
-            PlasmaVesselDistance(eq, surf),
+            PlasmaVesselDistance(eq, surf, jac_chunk_size=2),
             MagneticWell(eq),
         ],
         deriv_mode="blocked",
+        jac_chunk_size=10,
         use_jit=False,
     )
-    obj3 = ObjectiveFunction(
-        [
-            PlasmaVesselDistance(eq, surf),
-            MagneticWell(eq),
-        ],
-        deriv_mode="looped",
-        use_jit=False,
+    with pytest.warns(DeprecationWarning, match="looped"):
+        obj3 = ObjectiveFunction(
+            [
+                PlasmaVesselDistance(eq, surf),
+                MagneticWell(eq),
+            ],
+            deriv_mode="looped",
+            use_jit=False,
+        )
+    with pytest.warns(UserWarning, match="jac_chunk_size"):
+        obj1.build()
+    with pytest.warns(UserWarning, match="jac_chunk_size"):
+        obj2.build()
+    # check that default size works for blocked
+    assert obj2.objectives[1]._jac_chunk_size == np.ceil(
+        sum(t.dim_x for t in obj2.objectives[1].things) / 4
```

Comment:
Check my above comment. A user may want to create `obj2` with `jac_chunk_size=10` for the not specified sub_objectives. It might be tedious for them to write `jac_chunk_size=10` for each one of them in case they have a lot of objectives. I guess for now we don't have high number of objectives for a single optimization but it might be good for future.

Reply:
I am fine with the current way, I think it would be more confusing if we did not make it completely manual to specify